website/project/model.py

```diff
@@ -3070,6 +3072,20 @@ def manage_contributors(self, user_dicts, auth, save=False):
             if to_remove or permissions_changed and ['read'] in permissions_changed.values():
                 project_signals.write_permissions_revoked.send(self)
 
+    def throttle_add_contributor(self, contributor, permissions=None,
+                                 visible=True, auth=None, log=True, save=False):
+
+        if not throttle_period_expired(auth.user.email_last_sent,
```

Comment:
What happens here if auth isn't sent in? Probably need to add an `auth is not None` to the beginning of that or decorate the function with whatever decorators we have to ensure that the user has permission to do this.